server/socket/socket.js

```diff
@@ -1,54 +1,97 @@
-const connectedUsers = {};
-
+const socketToUsers = new Map();
+const usersToSockets = new Map();
+const rooms = {};
 module.exports = (server) => {
-  // const http = require('http').createServer(server);
-  const io = require('socket.io')(server, { origins: '*:*' });
+  const io = require('socket.io')(server, { origins: '*:*', path: '/sockets' });
   io.on('connection', (socket) => {
-    let connectedUser;
-    let room;
-    let isRoomFull = false;
+    const { id } = socket;
+    console.log('LOGGED IN! SOCKET ID', socket.id);
 
-    const waitingRoomFull = () => {
-      return Object.keys(connectedUsers[room]).length === 2
-    }
+    socket.on('check_rooms', ({ userId }) => {
+      console.log(userId);
+    });
 
-    socket.on('join_room', ({ user, roomId }) => {
-      connectedUser = user._id;
-      room = roomId;
+    socket.on('get_partner_name', ({ firstName, lastName, roomId }) => {
+      socket.broadcast
+        .to(roomId)
+        .emit('set_partner_name', { firstName, lastName });
+    });
+
+    socket.on('start_interview', (roomId) => {
+      socket.broadcast.to(roomId).emit('join_interview_room');
+      console.log('STARTING INTERVIEW, ROOMS @ ROOMID: ', {
+        users: rooms[roomId],
+      });
+    });
 
+    socket.on('create_room', ({ user, roomId }) => {
+      if (rooms[roomId]) {
+        if (Object.keys(rooms[roomId]).length < 2) {
+          console.log('joining room: ', roomId);
 
-      if (connectedUsers[room]) {
-        if (waitingRoomFull()) {
-          isRoomFull = true;
-          socket.emit('users', 'full');
-          return;
+          rooms[roomId][user._id] = user;
+          socket.roomId = roomId;
+          socket.join(roomId);
+        } else {
+          // room is full
+          io.to(socket.id).emit('room_full', null);
         }
-        connectedUsers[room][connectedUser] = user;
       } else {
-        user['isOwner'] = true
-        connectedUsers[room] = {
-          [connectedUser]: user
-        }
+        const userId = user._id;
+        user.isOwner = true;
```

Comment:
Is user a MongoDB object? If so you will need to save it for the data to persist

Reply:
This is not, it's a simple tag that gets imprinted on the object sent back from the socket "create_room" emit.  Its to render (Owner) next to the user who created the interview.  Lobby only logic.